Begin versioning spacefish releases

matchai opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
To help in the process of debugging issues, and to better document development progress, we will begin versioning releases.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->
- [ ] I have tested using **MacOS**
- [ ] I have tested using **Linux**

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have checked that no other PR duplicates mine
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

matchai committed:
Fix test npm script

salmanulfarzy commented on scripts/version.fish:
```diff
@@ -0,0 +1,9 @@
+set -l new_version (grep -E '"version": "v?([0-9]+\.){1,}' package.json | cut -d\" -f4 2> /dev/null)
+set -l filename $PWD/fish_prompt.fish
+
+sed -i -e "s/set -g SPACEFISH_VERSION .*/set -g SPACEFISH_VERSION $new_version/g" $filename
```
`sed -i` isn't POSIX compliant and behaves differently on BSD `sed` and GNU `sed`. See https://github.com/denysdovhan/spaceship-prompt/pull/430

matchai replied:
Good catch! 😄 
Will update appropriately.

matchai committed:
Add versioning to spacefish
- Add global variable with spacefish version (SPACEFISH_VERSION)
- Add "version" script to update global variable

matchai committed:
Merge branch 'master' into begin-versioning

matchai committed:
Use POSIX compliant sed for updating version
